data/json/effects_on_condition/dream_eocs.json

```
@@ -5,6 +5,108 @@
     "//": "used to prevent multiple dreams at the same sleep",
     "effect": [ { "math": [ "dream_counter", "=", "0" ] } ]
   },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_TROUBLED_SLEEP_FATIGUE_MESSAGES",
+    "//": "Prints a flavour text message to the log about being afraid of sleep if the character has entered the tired status, has the trubbled sleeper trait, and has not taken preventative medication. The EOC’s set to recur every 8 to 20 minutes.",
+    "global": true,
+    "recurrence": [ 480, 1200 ],
+    "condition": {
+      "and": [
+        { "math": [ "u_val('fatigue')", ">", "191" ] },
+        { "u_has_trait": "bad_dreams" },
+        { "not": { "u_has_effect": "took_prozac" } }
+      ]
+    },
+    "effect": [ { "u_message": "troubled_sleep_fatigue_messages", "snippet": true, "type": "bad" } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_TROUBLED_SLEEP_MESSAGE",
+    "//": "prints a message to the log upon the character falling asleep with the troubled sleeper trait. The text changes based on whether the player has taken preventative medication or not. Primarily intended for immersion and to let the player know if their medication, should they have taken it, is going to work to repress nightmares.",
+    "eoc_type": "EVENT",
+    "required_event": "character_falls_asleep",
+    "condition": { "and": [ { "u_has_trait": "bad_dreams" }, { "not": { "u_has_effect": "took_prozac" } } ] },
+    "effect": [
+      {
+        "u_message": "As your eyes close, unease brews in your mind, anxiety swirling in your gut.  You know the nightmares won't leave you at peace.",
+        "type": "bad"
+      }
+    ],
+    "false_effect": {
+      "run_eocs": [
+        {
+          "id": "EOC_TROUBLED_SLEEP_SAFE",
+          "condition": { "and": [ { "u_has_trait": "bad_dreams" }, { "u_has_effect": "took_prozac" } ] },
+          "effect": {
+            "u_message": "Surrendering yourself to the arms of sleep, you hope your medication will let you have a good night’s rest.",
+            "type": "good"
+          }
+        }
+      ]
+    }
+  },
```

This one is good, but can be simplified
```suggestion
    "condition": { "u_has_trait": "bad_dreams" },
    "effect": [
      {
        "if": { "u_has_effect": "took_prozac" },
        "then": {
          "u_message": "Surrendering yourself to the arms of sleep, you hope your medication will let you have a good night’s rest.",
          "type": "good"
        },
        "else": {
          "u_message": "As your eyes close, unease brews in your mind, anxiety swirling in your gut.  You know the nightmares won't leave you at peace.",
          "type": "bad"
        }
      }
    ]
  }
```